File: azure-active-directory-server/src/main/java/org/jetbrains/teamcity/aad/AADSchemeProperties.java
package org.jetbrains.teamcity.aad;

import jetbrains.buildServer.serverSide.auth.AuthModule;
import jetbrains.buildServer.serverSide.auth.LoginConfiguration;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.Map;

/**
 * @author Evgeniy.Koshkin
 */
public class AADSchemeProperties {

  @NotNull private final LoginConfiguration myLoginConfiguration;

  public AADSchemeProperties(@NotNull final LoginConfiguration loginConfiguration) {
    myLoginConfiguration = loginConfiguration;
  }

  @Nullable
  public String getAppTenantId() {
    final Map<String, String> properties = getAADSchemeProperties();
    return properties == null ? null : properties.get(AADConstants.AAD_TENANT_ID);
  }

  @Nullable
  public String getClientId() {
    final Map<String, String> properties = getAADSchemeProperties();
    return properties == null ? null : properties.get(AADConstants.CLIENT_ID_SCHEME_PROPERTY_KEY);
  }

  public boolean isSchemeConfigured() {
    return !myLoginConfiguration.getConfiguredAuthModules(AADAuthenticationScheme.class).isEmpty();
  }

  @Nullable
  private Map<String, String> getAADSchemeProperties() {
    final List<AuthModule<AADAuthenticationScheme>> aadAuthModules = myLoginConfiguration.getConfiguredAuthModules(AADAuthenticationScheme.class);
    return aadAuthModules.isEmpty() ? null : aadAuthModules.get(0).getProperties();
  }
}

File: azure-active-directory-server/src/main/kotlin/org/jetbrains/teamcity/aad/AADSchemeProperties.kt
/*
 * Copyright 2000-2020 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.jetbrains.teamcity.aad

import jetbrains.buildServer.serverSide.auth.LoginConfiguration

/**
 * @author Evgeniy.Koshkin
 */
class AADSchemeProperties(private val loginConfiguration: LoginConfiguration) {

    val appOAuthEndpoint: String?
        get() = properties?.get(AADConstants.AUTH_ENDPOINT_SCHEME_PROPERTY_KEY)

    val clientId: String?
        get() = properties?.get(AADConstants.CLIENT_ID_SCHEME_PROPERTY_KEY)

    val authPrompt: String?
        get() = properties?.get(AADConstants.AUTH_PROMPT)

    val isSchemeConfigured: Boolean
        get() = properties != null

    private val properties: Map<String, String>?
        get() {
            val aadAuthModules = loginConfiguration.getConfiguredAuthModules<AADAuthenticationScheme>(AADAuthenticationScheme::class.java)
            return if (aadAuthModules.isEmpty()) null else aadAuthModules[0].properties
        }
}
